File: src/main/java/nl/haltedata/chb/dto/ChbQuay.java
package nl.haltedata.chb.dto;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(schema = "chb")
@Getter
@Setter
@EqualsAndHashCode(of = "id")
public class ChbQuay {

    @Id
    private String id;
    private String stopPlaceId;
    private String stopPlaceName;
    private String stopPlaceLongName;
    private List<String> transportModes;
    private Boolean onlygetout;
    private String quayCode;
    private String quayName;
    private String stopSideCode;
    private LocalDateTime validfrom;
    private LocalDateTime mutationdate;
    private String quayType;
    private String quayStatus;
//    private Point rdLocation;
//    private Point wgsLocation;
    private Integer bearing;
    private String town;
    private String level;
    private String street;
    private String location;  
}

File: frontend/src/app/quays/chb-quay.ts
export default class ChbQuay {
    id!: string;
    stopPlaceId!: string;
    stopPlaceName!: string;
    stopPlaceLongName!: string;
    transportModes!: string[];
    onlygetout!: boolean;
    quayCode!: string;
    quayName!: string;
    stopSideCode!: string;
    validfrom!: Date;
    mutationdate!: Date;
    quayType!: string;
    quayStatus!: string;
//    private Point rdLocation;
//    private Point wgsLocation;
    bearing!: number;
    town!: string;
    level!: string;
    street!: string;
    location!: string;
}
